imports/plugins/core/accounts/server/no-meteor/mutations/inviteShopMember.js

```diff
@@ -0,0 +1,159 @@
+import _, { difference } from "lodash";
+import SimpleSchema from "simpl-schema";
+import Random from "@reactioncommerce/random";
+import ReactionError from "@reactioncommerce/reaction-error";
+import { Meteor } from "meteor/meteor";
+import { Accounts as MeteorAccounts } from "meteor/accounts-base";
+import Reaction from "/imports/plugins/core/core/server/Reaction";
+import getCurrentUserName from "/imports/plugins/core/accounts/server/no-meteor/util/getCurrentUserName";
+import getDataForEmail from "/imports/plugins/core/accounts/server/util/getDataForEmail";
+
+const inputSchema = new SimpleSchema({
+  email: String,
+  groupId: String,
+  name: String,
+  shopId: String
+});
+
+/**
+ * @name accounts/inviteShopMember
+ * @memberof Mutations/Accounts
+ * @summary Invite new admin users (not consumers) to secure access in the dashboard to permissions
+ * as specified in packages/roles
+ * @param {Object} context - GraphQL execution context
+ * @param {Object} input - Necessary input for mutation. See SimpleSchema.
+ * @param {String} input.shopId - shop to invite user
+ * @param {String} input.groupId - groupId to invite user
+ * @param {String} input.email - email of invitee
+ * @param {String} input.name - name of invitee
+ * @return {Promise<Object>} with boolean of found new account === true || false
+ */
+export default async function inviteShopMember(context, input) {
+  inputSchema.validate(input);
+  const { appEvents, collections, user: userFromContext, userHasPermission, userId: userIdFromContext } = context;
+  const { Accounts, Groups, Shops, users } = collections;
+  const {
+    email,
+    groupId,
+    name,
+    shopId
+  } = input;
+
+  if (!userHasPermission(["reaction-accounts"], shopId)) {
+    throw new ReactionError("access-denied", "Access denied");
+  }
+
+  // we always use primary shop data, so retrieve this shop first with `Reaction` helper,
+  // and only query the `Shops` collection if shopId !== primaryShop._id
+  const primaryShop = Reaction.getPrimaryShop();
+  if (!primaryShop) throw new ReactionError("not-found", "No primary shop found");
+
+  let shop = primaryShop;
+  if (primaryShop._id !== shopId) {
+    shop = await Shops.findOne({ _id: shopId });
+  }
+  if (!shop) throw new ReactionError("not-found", "No shop found");
+
+  const group = await Groups.findOne({ _id: groupId });
+  if (!group) throw new ReactionError("not-found", "No gr found");
+
+  // we don't allow direct invitation of "owners", throw an error if that is the group
+  const ownerGroup = await Groups.findOne({ name: "owner" });
+  if (group._id === ownerGroup._id) {
+    throw new ReactionError("bad-request", "Cannot directly invite owner");
+  }
```

Comment:
I believe you could avoid the second `findOne` here by changing this to `if (group.name === "owner")`

Reply:
Yeah... I originally had that, but then when I added the permissions check on line 71, I was doing the extra look up anyway, so decided to move it up a bit and change this check.

But now that that second permission check will be removed, I'll revert it back to what you have commented here.